src/internal/packager/images/pull.go

```diff
@@ -448,39 +486,10 @@ func (i *ImageConfig) PullImage(src string, spinner *message.Spinner) (img v1.Im
 		// If crane is unable to pull the image, try to load it from the local docker daemon.
 		message.Debugf("crane unable to pull image %s: %s", src, err)
 		spinner.Updatef("Falling back to docker for %s. This may take some time.", src)
-
-		// Parse the image reference to get the image name.
-		reference, err := name.ParseReference(src)
-		if err != nil {
-			return nil, false, fmt.Errorf("failed to parse image reference %s: %w", src, err)
-		}
-
-		// Attempt to connect to the local docker daemon.
-		ctx := context.TODO()
-		cli, err := client.NewClientWithOpts(client.FromEnv)
-		if err != nil {
-			return nil, false, fmt.Errorf("docker not available: %w", err)
-		}
-		cli.NegotiateAPIVersion(ctx)
-
-		// Inspect the image to get the size.
-		rawImg, _, err := cli.ImageInspectWithRaw(ctx, src)
-		if err != nil {
-			return nil, false, fmt.Errorf("failed to inspect image %s via docker: %w", src, err)
-		}
-
-		// Warn the user if the image is large.
-		if rawImg.Size > 750*1000*1000 {
-			warn := pterm.DefaultParagraph.WithMaxWidth(message.TermWidth).Sprintf("%s is %s and may take a very long time to load via docker. "+
-				"See https://docs.zarf.dev/docs/faq for suggestions on how to improve large local image loading operations.",
-				src, utils.ByteFormat(float64(rawImg.Size), 2))
-			spinner.Warnf(warn)
-		}
-
-		// Use unbuffered opener to avoid OOM Kill issues https://github.com/defenseunicorns/zarf/issues/1214.
-		// This will also take for ever to load large images.
-		if img, err = daemon.Image(reference, daemon.WithUnbufferedOpener()); err != nil {
-			return nil, false, fmt.Errorf("failed to load image %s from docker daemon: %w", src, err)
+		var localErr error
+		img, localErr = i.pullLocalImage(src, spinner)
+		if localErr != nil {
+			return nil, false, fmt.Errorf("remote repository: %w, local filesystem: %w", err, localErr)
```

Comment:
One displaying both errors I think it would be better to handle this in the fallback logic as two separate messages and follow the git fallback pattern more closely - the retry logic may also be something to look into if we want to show simpler messages in between (debugging the full error) with the last error(s) showing both.

Git Fallback:
![image](https://github.com/defenseunicorns/zarf/assets/3977569/3893a765-4a05-4f31-b6c9-31458fa9cf6e)

Current implementation:
![image](https://github.com/defenseunicorns/zarf/assets/3977569/26cf1f19-54fd-44bb-bf90-ec93937956f1)

Reply:
Simplified messaging and broke up lines visually
![image](https://github.com/defenseunicorns/zarf/assets/3977569/17970ea2-5ab0-4441-9db0-f6b3988ecc25)